### PR TITLE
[doc] Add an example of batch deleting data in java-api.md.

### PR DESCRIPTION
**Pull request by davedwwang**
Add an example of batch deleting data

### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: docs/content/program-api/java-api.md [hunk 1]
```diff
@@ -725,3 +725,110 @@ public class StreamWriteTable {
...
+        builder.option(CoreOptions.BUCKET.key(), "3");
+        Schema schema = builder.build();
+        if (catalog.tableExists(tableId)) {
+            catalog.dropTable(tableId, true);
+        }
+        catalog.createTable(tableId, schema, false);
+        return catalog.getTable(tableId);
+    }
+
+    public void deletePartition(Table t) throws Exception {
```
Comment (JingsongLi) [inline docs/content/program-api/java-api.md:791]: Can you just add in `catalog-api`?

File: docs/content/program-api/java-api.md [hunk 2]
```diff
@@ -725,3 +725,110 @@ public class StreamWriteTable {
...
+        for (BinaryRow r:binaryRows) {
+            BinaryString dt = r.getString(0);
+            partitions.add(Collections.singletonMap("dt", dt.toString()));
+        }
+        try(FileStoreCommit storeCommit = storeTable.store().newCommit(createCommitUser(storeTable.coreOptions().toConfiguration()))) {
+            storeCommit.dropPartitions(partitions, Long.MAX_VALUE);
+        }
+    }
+
+    public void deleteRecord(Table t) throws Exception {
```
Comment (JingsongLi) [inline docs/content/program-api/java-api.md:808]: Maybe just add this in `Batch Write`.

File: docs/content/program-api/java-api.md (hunk 1 above)
Comment (davedwwang) [reply docs/content/program-api/java-api.md:791]: OK

**Comment by JingsongLi** [comment]
Close inactive PR, feel free to re-open it

**Pull request closed without merging**